magento/magento2
# Fix horizontal media gallery thumbnails pixelated

Fixes issue #23877 that causes horizontal media gallery thumbnails to be oversized.

### Description (*)
Changed the gallery.less file so the horizontal_ratio images don't get a `height: 100%` styling anymore. This causes the width of the image to expand and making it bigger than the image actually is. Also introduced the `.translate` mixin to combine `translateX` and `translateY` styles so the non-resized thumbnail can be center aligned.

### Fixed Issues (if relevant)
1. magento/magento2#23877: Media gallery thumbnails stretched when image is horizontal ratio

### Manual testing scenarios (*)
1. Have a product with images that have a horizontal ratio (the width is bigger than the height)
2. Make sure that `product_image_white_borders` is set to 0 in the view.xml of the activated theme
3. If cache images are already created remove the `pub/media/catalog/product/cache/` folder.
4. View the product page to see if the images are rendered correctly.

### Questions or comments
As far as I can see this doesn't influence the image rendering if `product_image_white_borders` is enabled. If we only want to apply this to the image when `product_image_white_borders` is disabled we have to introduce a new class to apply the correct styles to.

### Contribution checklist (*)
